sdk/core/azure-core/inc/http/http.hpp

```diff
@@ -214,21 +238,33 @@ class Response
 
 public:
   Response(uint16_t statusCode, std::string const& reasonPhrase)
-      : Response(statusCode, reasonPhrase, http::BodyBuffer::null, http::BodyStream::null)
+      : Response(statusCode, reasonPhrase, std::vector<uint8_t>(), http::BodyStream::null)
   {
   }
 
+  // default constructor for bulding from base response
+  Response() : Response(0, "", std::vector<uint8_t>(), http::BodyStream::null) {}
+
   // Methods used to build HTTP response
+  void setVersion(uint16_t mayorVersion, uint16_t minorVersion)
+  {
+    m_httpVersion = http::HttpVersion(mayorVersion, minorVersion);
+  }
+  void setStatusCode(uint16_t statusCode) { m_statusCode = statusCode; }
+  void setReasonPhrase(std::string reasonPhrase) { m_reasonPhrase = reasonPhrase; }
   void addHeader(std::string const& name, std::string const& value);
-  void setBody(BodyBuffer* bodyBuffer);
   void setBody(BodyStream* bodyStream);
 
+  void appendBody(uint8_t* ptr, uint64_t size);
+
   // Methods used by transport layer (and logger) to send response
   uint16_t getStatusCode();
   std::string const& getReasonPhrase();
+  std::string getHttpVersion();
   std::map<std::string, std::string> const& getHeaders();
   http::BodyStream* getBodyStream();
-  http::BodyBuffer* getBodyBuffer();
+  std::vector<uint8_t>& getBodyBuffer();
+  std::string getStringBody();
```

Comment:
This does not belong here because a Response object cannot guarantee to a caller that its body is a string. 
Only the calling code has the knowledge to know whether the body is supposed to be a string or not and so, that calling code should get the body and then decide to treat it as a string.